src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -629,73 +632,74 @@ private void SetDefaultProjectName()
 
         private async Task EnsureNuGetAndVsProjectAdapterCacheAsync()
         {
-            ThreadHelper.ThrowIfNotOnUIThread();
-
-            if (!_cacheInitialized && IsSolutionOpen)
+            await _initLock.ExecuteNuGetOperationAsync(async () =>
```

Comment:
Project Added/Renamed will skip by https://github.com/NuGet/NuGet.Client/pull/1604/files#diff-829b08c28853b1bf9bf136e579c869d1L586, they will not wait on lock